src/core/src/pass/constant_folding.cpp

```diff
@@ -23,9 +25,9 @@ using namespace std;
  *
  * \return true if output can be folded otherwise false.
  */
-const auto is_output_foldable = [](const ov::Output<ov::Node>& output) {
+const auto is_output_foldable = [](const ov::Output<ov::Node>& output, const std::string& key = "can_be_folded") {
     const auto& rt_info = output.get_node()->get_rt_info();
-    return !rt_info.count("can_be_folded") || rt_info.at("can_be_folded").as<bool>();
+    return rt_info.count(key) && rt_info.at(key).as<bool>();
```

Comment:
Is this check ok?
as for default `key` value is not same as before modification there is no negation in first check `!rt_info.count("can_be_folded")` -> `rt_info.count(key)`

Reply:
PR is updated, the comment is not relevant

Reply:
Different approach was chosen. Resolving the comment.